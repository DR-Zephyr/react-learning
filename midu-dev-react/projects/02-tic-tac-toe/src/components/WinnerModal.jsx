import { Square } from './Square';
import PropTypes from 'prop-types';

export function WinnerModal({ winner, resetGame }) {
    //
    const winnerText = winner === false ? 'Empate' : `Ganador:`;
    //
    if (winner === null) return null;

    return (
        <section className='winner'>
            <div className='text'>
                <h2>{winnerText}</h2>

                <header className='win'>
                    {winner && <Square>{winner}</Square>}
                </header>

                <button onClick={resetGame}>Empezar de nuevo</button>
            </div>
        </section>
    );
}

WinnerModal.propTypes = {
    winner: PropTypes.string,
    resetGame: PropTypes.func.isRequired,
};
