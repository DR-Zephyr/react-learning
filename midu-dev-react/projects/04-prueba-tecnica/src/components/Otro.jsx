//
import { useCatImage, useCatFact } from '../hooks'

export const Otro = () => {
  const { fact, refreshFact } = useCatFact()
  const { imageUrl } = useCatImage({ fact })

  return (
    <>
      {imageUrl && (
        <img
          src={imageUrl}
          alt='Image extracted from the cat fact'
        />
      )}
    </>
  )
}
