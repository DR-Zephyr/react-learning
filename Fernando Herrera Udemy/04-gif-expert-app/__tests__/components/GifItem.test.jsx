import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en el GifItem', () => {
    //
    const title = 'Saitma',
        urlImage = 'https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/source';
    //
    test('Debe hacer match con el snapshot', () => {
        const { asFragment } = render(
            <GifItem
                title={title}
                url={urlImage}
            />
        );
        //
        expect(asFragment()).toMatchSnapshot();
    });
    //
    test('debe mostar la imagen con el src y el alt indicado', () => {
        //
        render(
            <GifItem
                title={title}
                url={urlImage}
            />
        );
        //
        const { src, alt } = screen.getByRole('img');
        //
        expect(src).toBe(urlImage);
        expect(alt).toBe(title);
    });
    //
    test('Debe de mostar el titulo en el componente', () => {
        render(
            <GifItem
                title={title}
                url={urlImage}
            />
        );
        //
        expect(screen.getByText(title)).toBeTruthy();
    });
});
