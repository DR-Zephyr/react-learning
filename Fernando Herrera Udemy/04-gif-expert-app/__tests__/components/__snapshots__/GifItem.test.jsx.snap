// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifItem Debe hacer match con el snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <img
      alt="Saitma"
      src="https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/source"
    />
    <p>
      Saitma
    </p>
  </div>
</DocumentFragment>
`;
